File: fortran/pr006.f90
! Problem 6
!
! 14 December 2001
!
! The sum of the squares of the first ten natural numbers is,
!
! 1^2 + 2^2 + ... + 10^2 = 385
!
! The square of the sum of the first ten natural numbers is,
!
! (1 + 2 + ... + 10)^2 = 55^2 = 3025
!
! Hence the difference between the sum of the squares of the first ten
! natural numbers and the square of the sum is 3025 − 385 = 2640.
!
! Find the difference between the sum of the squares of the first one
! hundred natural numbers and the square of the sum.
! 25164150

subroutine pr006

  implicit none
  integer :: i, sumsq, sqsum

  sumsq = 0
  sqsum = 0
  do i = 1, 100
    sumsq = sumsq + i**2
    sqsum = sqsum + i
  end do
  sqsum = sqsum**2

  print *, sqsum - sumsq

end subroutine pr006

File: kotlin/src/pr006.kt
package org.davidb.euler

// Problem 6
//
// 14 December 2001
//
// The sum of the squares of the first ten natural numbers is,
//
// 1^2 + 2^2 + ... + 10^2 = 385
//
// The square of the sum of the first ten natural numbers is,
//
// (1 + 2 + ... + 10)^2 = 55^2 = 3025
//
// Hence the difference between the sum of the squares of the first ten
// natural numbers and the square of the sum is 3025 − 385 = 2640.
//
// Find the difference between the sum of the squares of the first one
// hundred natural numbers and the square of the sum.
// 25164150

// define_problem!(pr006, 6, 25164150);

class Pr006: EulerSolution<Int>() {
    override val answer = 25164150
    override val num = 6

    override fun solve(): Int {
        val sum_sq = (1 .. 100).map { it * it }.sum()
        val tmp = (1 .. 100).sum()
        val sq_sum = tmp * tmp

        return sq_sum - sum_sq
    }
}
